src/Microsoft.Azure.WebJobs.Logging/LogFactory.cs

```diff
@@ -15,23 +13,35 @@ public static class LogFactory
         /// <summary>
         /// Get a reader that reads from the given table. 
         /// </summary>
-        /// <param name="table"></param>
+        /// <param name="tableLookup"></param>
         /// <returns></returns>
-        public static ILogReader NewReader(CloudTable table)
+        public static ILogReader NewReader(ILogTableProvider tableLookup)
         {
-            return new LogReader(table);
+            return new LogReader(tableLookup);
         }
 
         /// <summary>
         /// Create a new writer for the given compute container name that writes to the given table.
         /// Multiple compute instances can write to the same table simultaneously without interference. 
         /// </summary>
         /// <param name="computerContainerName">name of the compute container. Likley %COMPUTERNAME%. </param>
-        /// <param name="table">underlying azure storage table to write to.</param>
+        /// <param name="tableLookup">underlying azure storage table to write to. 
+        /// Passed a string suffix (which will consist of valid Azure table characters). 
+        /// This must be a deterministic (replayable) function.</param>
         /// <returns></returns>
-        public static ILogWriter NewWriter(string computerContainerName, CloudTable table)
+        public static ILogWriter NewWriter(string computerContainerName, ILogTableProvider tableLookup)
+        {
+            return new LogWriter(computerContainerName, tableLookup);
+        }
+
+        /// <summary>
+        /// Get a default table provider for the given tableClient. This will generate table names with the given prefix.
+        /// </summary>
+        /// <param name="tableClient">storage client for where to generate tables</param>
+        /// <param name="tableNamePrefix">prefix for tables to generate. This should be a valid azure table name.</param>
+        public static ILogTableProvider NewTableProvider(CloudTableClient tableClient, string tableNamePrefix = LogFactory.DefaultLogTableName)
         {
-            return new LogWriter(computerContainerName, table);
+            return new EpochTableProvider(tableClient, tableNamePrefix);
```

Comment:
ILogTableProvider is public so people can implement instances of it, but they can't plug in their implementations into our infrastructure, right? I'm not saying that is a goal, just trying to understand the provider interface and your intent for it.